blockchain/contracts/FlatNFTtheWorld.sol

```diff
@@ -1399,6 +1405,29 @@ contract NFTtheWorld {
         }
     }
 
+
+    // to be called automatically from backend
+    // checks weither reservation has timed out & if so, if reservedFor != owner, meaning it wasnt bought
+    // if so, reinstate as if Drop was executed but NFT wasnt reserved 
+    function revertTimedoutReservations(uint256 _dropHash) public returns (uint256){
+        uint256 reservationsReverted =0;
+        for(uint256 i;i<nftOwnerships[_dropHash].length;i++){
+            if(nftOwnerships[_dropHash][i].reservedUntil>=0 &&
+                nftOwnerships[_dropHash][i].reservedUntil<=block.timestamp && 
+                nftOwnerships[_dropHash][i].owner!=nftOwnerships[_dropHash][i].reservedFor){
+                nftReservationInformationOfUsers[nftOwnerships[_dropHash][i].reservedFor][_dropHash].pop();
+                nftOwnerships[_dropHash][i].reservedUntil =0;
+                reservedNFTsCount[_dropHash] -= nftReservations[nftOwnerships[_dropHash][i].reservedFor][_dropHash];
+                nftReservations[nftOwnerships[_dropHash][i].reservedFor][_dropHash] = 0;
+                nftOwnerships[_dropHash][i].reservedFor=nftOwnerships[_dropHash][i].owner;
+                availableNFTs[_dropHash].push(nftOwnerships[_dropHash][i].uri);
+                reservationsReverted++;
+            }
+
+        }
+        return reservationsReverted;
```

Comment:
I think if this is not needed for anything we should get rid of it

Reply:
I think if we were to call this function, getting to know how many NFTs have been put back to distribute would be beneficial information. But isnt needed for now